Stop, drop, and roll + Plasma fire damage tweaks

Semuca opened the pull request:
### Purpose
Reworked stopping, dropping, and rolling (Ticks the third box off #6398) and tweaks damage done by plasma fires to make it not too deadly.

### Notes:
A stop drop and roll works like this:

![ezgif-7-fd2a5afb4c70](https://user-images.githubusercontent.com/62051810/116661223-b29a2f80-a9c6-11eb-919a-d0077bff72a6.gif)

(Excuse the overlays, none of them seem to work at the moment)

When you click the resist button while on fire (The gif shows me clicking the drop button, but that's a recording error) you'll start a coroutine to drop and roll, reducing some fire stacks every roll until the coroutine ends or you die. After that, it'll automatically stand you up. There's also some text once you begin the roll, but I added after the gif and couldn't be bothered to re-record.

I've also tweaked the plasma fire damage so it's not ridiculous. It now does a quarter of the damage it did before, so it doesn't kill you in 3 seconds and gives people more of a chance to get away.

### Changelog:
CL: Stop, drop, and roll is now better
CL: Plasma fire damage tweaks

Semuca committed:
Stop, drop, and roll + plasma fire damage tweaks

NoooneyDude commented on UnityProject/Assets/Scripts/Player/PlayerNetworkActions.cs:
```diff
@@ -134,45 +136,11 @@ public void CmdResist()
+				playerScript.gameObject,
+				$"You drop to the ground and frantically try to put yourself out!",
+				playerScript.playerName + " is trying to extinguish themself");
+			StartCoroutine("Roll");
```
```suggestion
			StartCoroutine(nameof(Roll));
```

Semuca replied:
Changed

NoooneyDude commented on UnityProject/Assets/Scripts/Player/PlayerNetworkActions.cs:
```diff
@@ -134,45 +136,11 @@ public void CmdResist()
+			Chat.AddActionMsgToChat(
+				playerScript.gameObject,
+				$"You drop to the ground and frantically try to put yourself out!",
+				playerScript.playerName + " is trying to extinguish themself");
```
```suggestion
				"You drop to the ground and frantically try to put yourself out!",
				$"{playerScript.playerName} is trying to extinguish themself!");
```

NoooneyDude commented on UnityProject/Assets/Scripts/Player/PlayerNetworkActions.cs:
```diff
@@ -192,6 +160,86 @@ public void CmdResist()
+				case OrientationEnum.Left:
+					faceDir = Orientation.Up;
+					break;
+			}
```
I see you've just moved this code, but have you considered using `Orientation.Rotate()` instead of converting to `OrientationEnum` and back? Or does it not rotate in the right order? Would need to try it to see.

Semuca replied:
No, I didn't know it existed. I'll give it a shot now though

Semuca committed:
Style guide changes

Semuca committed:
Guarding against stun cases and cleaning up rolling code

Also made rolling have an effect on certain actions, like being helped up.

Cleaned up some StandardProgressAction code while I was at it.

Semuca committed:
begone codacy

Semuca committed:
One more style guide update

removing '== true'
